contracts/CumulativeMerkleDrop.sol

```diff
@@ -15,25 +13,39 @@ contract CumulativeMerkleDrop is Ownable, ICumulativeMerkleDrop {
     using MerkleProof for bytes32[];
 
     address public immutable override token;
+    address public rewardsHolder;
 
     bytes32 public override merkleRoot;
     mapping(address => uint256) public cumulativeClaimed;
+    struct Claim {
+        address account;
+        uint256 amount;
+        bytes32[] proof;
+    }
 
-    constructor(address token_) {
+    event RewardsHolderUpdated(address oldRewardsHolder, address newRewardsHolder);
+
+    constructor(address token_, address rewardsHolder_) {
         token = token_;
+        rewardsHolder = rewardsHolder_;
     }
 
     function setMerkleRoot(bytes32 merkleRoot_) external override onlyOwner {
         emit MerkelRootUpdated(merkleRoot, merkleRoot_);
         merkleRoot = merkleRoot_;
     }
 
+    function setRewardsHolder(address rewardsHolder_) external onlyOwner {
+        emit RewardsHolderUpdated(rewardsHolder, rewardsHolder_);
+        rewardsHolder = rewardsHolder_;
```

Comment:
Maybe add address check here too? maybe not only for zero address but that some tokens belong to that address or even allowance? wdyt?

Reply:
i think checking that it's not 0 address is a good idea. I think checking for the allowance/balance might be premature - it's possible that those would only be sorted afterwards.